pkg/kubelet/pod_workers.go

```diff
@@ -79,11 +87,61 @@ func newPodWorkers(runtimeCache kubecontainer.RuntimeCache, syncPodFn syncPodFnT
 	}
 }
 
+// shouldBackOffMore returns true if the back off interval should be increased,
+// based on the the summary of the last sync.
+func shouldBackOffMore(summary *syncPodSummary) bool {
+	if summary.Unhealthy > 0 || summary.NotFound > 0 {
+		// Some containers were restarted in the last sync, we should increase
+		// the backoff interval. Note that we "NotFound" could also mean that
+		// the container was started the first time; the caller could rule out
+		// the possiblity by checking whether they are seeing the pod the first
+		// time. However, if we ever allow the addition/deletion of containers
+		// in in a pod, we need to revise this logic.
+		return true
+	}
+	return false
+}
+
+// computeNewBackoffTime computes and returns the new backoff interval,
+// along with the timestamp where the interval was last increased.
+func computeBackoffTime(summary *syncPodSummary, lastInterval time.Duration, firstSync bool,
```

Comment:
Yes, that's why this is a WIP ;)

I will add more tests later. Just wanted to make sure this is the direction that we wanted.